src/App.java

```diff
@@ -1,5 +1,182 @@
+     */
+    public void startMessage() {
+        printer.output("");
+        printer.output("Welcome to Flash Cards!");
+        printer.output("Please follow the instructions below to begin your study session.");
+        printer.output("If at any point you want to end your study session, please type “End study”");
+        printer.output("");
+        printer.output("Please pick a subject to study");
+        printer.output("(Please type the name of the text file)");
+    }
+
+    /**
+     * Saves the user input into a string that holds the file name 
+     * of the deck the user wants to use. If the input does not 
+     * include ".txt", the method adds it.
+     * @return user input as a string
+     */
+    public String pickFlashcard() {
+        String m = printer.input().toLowerCase();
+        if (m.indexOf(".txt") == -1){
+            return m + ".txt";
+        } else return m;
+    }
+
+    /**
+     * Takes in the string from pickFlashcard() as a parameter and opens the matching file.
+     * 
+     * @param filename - name of the file
+     * @return Deck object according to the corresponding file
+     */
+    public Deck readFlashcard(String filename) {
+        Deck m = new Deck();
+        String card = "";
+            try {
+                printer.openFile(filename);
+                while (printer.fileHasNextLine()) {
+                    card = printer.getNextLine();
+                    int i = card.indexOf(" - ");
+                    if (i == -1)
+                        break;
+                    String q = card.substring(0, i);
+                    String l = card.substring(i + 3);
+                    m.addWord(q);
+                    m.addDefinition(l);
+                }
+            } catch (Exception e) {
+                printer.output("");
+                printer.output(e.toString());
+                printer.output("Please try again - file did not exist");
+
+            }
+
+        return m;
+    }
+
+    /**
+     * Allows the user to choose the mode of study.
+     * @return int from 1-3 which corresponds to a mode of study.
+     */
+    public int chooseMode() {
+        printer.output("");
+        printer.output("Select a mode of study");
+        printer.output("(Please type the mode study as listed below)");
+        printer.output("1. Guess word");
+        printer.output("2. Guess definition");
+        printer.output("3. Mix");
+        printer.output("");
+        String answer = printer.input().toLowerCase();
+        printer.output("");
+        if (answer.equals("guess word")) {
+            return 1;
+        } else if (answer.equals("guess definition")) {
+            return 2;
+        } else if (answer.equals("mix")) {
+            return 3;
+        } else {
+            return -1;
+        }
+    }
+
+    /**
+     * Depending on the mode selected, this method will present to the user: definitions, words, or a mix of both.
+     */
+    public void doFlashcards() {

```

Comment:
this method is getting a little long, you should use submethods for each of the choices just to reduce the cognitive mode of reading any one particular method